File: abc/151/C.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var sc = bufio.NewScanner(os.Stdin)

func nextInt() int {
	sc.Scan()
	sc.Split(bufio.ScanWords)
	i, e := strconv.Atoi(sc.Text())
	if e != nil {
		panic(e)
	}
	return i
}

func nextLine() string {
	sc.Scan()
	return sc.Text()
}

func main() {
	var n, m int
	pmap := map[int]bool{}
	fmt.Scan(&n)
	fmt.Scan(&m)
	var num, ac, wa int
	var ans string
	for i := 0; i < m; i++ {
		num, ans = nextInt(), nextLine()
		val, ok := pmap[num]
		// まだ AC も WA もしてないとき
		if !ok {

			if ans == "AC" {
				pmap[num] = true
				ac++
			} else {
				pmap[num] = false
				wa++
			}
		} else {
			// 既に AC または WA してるとき
			if val {
				// AC 済み
			} else {
				// WA
				if ans == "AC" {
					// 正解した
					pmap[num] = true
					ac++
				} else {
					// ひきつづき不正解
					wa++
				}
			}
		}
	}
	fmt.Printf("%d %d\n", ac, wa)
}

File: abc/151/C.cpp
#include <cstdio>

int main() {
    int n, m;
    scanf("%d", &n);
    scanf("%d", &m);
    int num;
    int ac = 0, wa = 0;
    char ans[10];

    int WA[100007];
    bool AC[100007];
    for(int i = 1; i <= n; i++) {
        AC[i] = 0;
        WA[i] = 0;
    }
    for(int i = 0 ; i < m ; i++){
        scanf("%d", &num);
        scanf("%s", ans);
        if (AC[num]) {
            continue;
        }
        if (ans[0] == 'A') {
            AC[num] = true;
        } else { // if (ans[0] == 'W'){
            WA[num] += 1;
        }
    }
    for(int i = 1; i <= n; i++) {
        if(AC[i]) {
            ac++;
            wa += WA[i];
        }
    }
    printf("%d %d\n", ac, wa);
}
